packages/diagram/src/module/diagrams/content/uml/classifier/propertiesAndMethods.ts

```diff
@@ -0,0 +1,250 @@
+    ExecutableNumberLiteralExpression,
+    Expression,
+    fun,
+    FunctionObject,
+    functionType,
+    IdentifierExpression,
+    IndexExpression,
+    InterpreterModule,
+    InvocationExpression,
+    jsFun,
+    NumberLiteralExpression,
+    OperatorExpression,
+    parse,
+    RuntimeError,
+    StringLiteralExpression
+} from "@hylimo/core";
+
+/**
+ * Module providing the properties and methods content handler
+ * Requires the sections content handler
+ */
+export const propertiesAndMethodsModule = InterpreterModule.create(
+    "uml/classifier/propertiesAndMethods",
+    [],
+    [],
+    [
+        assign(
+            "_classifierEntryScopeGenerator",
+            fun([
+                ...parse(
+                    `
+                        (visibility, scope) = args
+                    `
+                ),
+                jsFun(
+                    (args, context) => {
+                        const scopeFunction = args.getLocalFieldOrUndefined(0)!.value;
+                        if (!(scopeFunction instanceof FunctionObject)) {
+                            throw new Error("scope is not a function");
+                        }
+                        const expressions = scopeFunction.definition.expressions.map(
+                            (expression) => expression.expression
+                        );
+                        const visibility = assertString(context.getField("visibility"));
+                        const scope = context.getField("scope");
+                        const [fields, functions] = convertFieldsAndFunctions(expressions);
+
+                        function addEntriesToScope(entries: string[], index: number): void {
+                            if (entries.length > 0) {
+                                scope.invoke(
+                                    [
+                                        ...entries.map((entry) => ({
+                                            value: new ExecutableConstExpression({
+                                                value: context.newString(visibility + entry)
+                                            })
+                                        })),
+                                        {
+                                            name: "section",
+                                            value: new ExecutableNumberLiteralExpression(undefined, index)
+                                        }
+                                    ],
+                                    context
+                                );
+                            }
+                        }
+                        addEntriesToScope(fields, 0);
+                        addEntriesToScope(functions, 1);
+
+                        return context.null;
+                    },
+                    {
+                        docs: `
+                            Function to take a function in which fields and functions can be declared declaratively.
```

Comment:
Same as one of the comments above

Reply:
changed with 9c9eecbbf2f389b6d1910ddee2e676b161f6e020